typescript/phoenix-sdk/src/trader.ts

```diff
@@ -59,39 +47,77 @@ export class Trader {
           tokenAccount.account.data,
           token.data.decimals
         );
-
-      // Subscribe to token balance updates
-      connection.onAccountChange(tokenAccount.pubkey, (accountInfo) => {
-        trader.tokenBalances[token.data.mintKey.toBase58()] =
-          getTokenAmountFromBuffer(accountInfo.data, token.data.decimals);
-      });
     }
 
     return trader;
   }
 
   /**
    * Refreshes the trader data
+   *
+   * @param connection The Solana `Connection` object
    */
-  async refresh() {
+  async refresh(connection: Connection) {
     // Refresh token balances
-    for (const tokenMint in this.tokenBalances) {
-      const mint = new PublicKey(tokenMint);
-      const tokenAccounts = await this.connection.getTokenAccountsByOwner(
-        mint,
+    for (const mintKey in this.tokenBalances) {
```

Comment:
here we should do `Promise.all` so we can kick off the network requests in parallel rather than in sequence